### PR TITLE
Unit test functions

### PR DESCRIPTION
**Pull request by briancmurphy87**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,16059 +0,0 @@
-{
```
Comment (ryanlabouve) [inline package-lock.json]: Can you do me a favor and run `npm install` again? I don't think we want to delete every line from `package-lock.json`. May have been  an npm glitch?

File: src/lib/utils/functions.test.ts [hunk 2]
```diff
@@ -1,5 +1,128 @@
...
+
+
+describe("evaluateNft", function () {
+
+  test('#evaluateNft', async () => {
+
+    const all_nft_examples = [g_lazy_lions_label, g_alpacadabra_label, g_devs_for_revolution_label];
+    // const all_nft_examples = [g_devs_for_revolution_label];
+
+    for(let ex_index = 0; ex_index < all_nft_examples.length; ex_index++) {
```
Comment (ryanlabouve) [inline src/lib/utils/functions.test.ts:87]: Minor note on style. Typically in js-land it's preferred to use a more functional (as opposed to imperative) style to iterate over loops. But this is fine for this test. 

e.g.
```js
[g_lazy_lions_label, g_alpacadabra_label, g_devs_for_revolution_label].forEach((nftExample) => {
   // test
});
```

File: src/lib/utils/functions.test.ts [hunk 3]
```diff
@@ -1,5 +1,128 @@
...
+      const nft_example = all_nft_examples[ex_index];
+      const token_uri = g_nft_examples[nft_example]["token_uri"]
+
+      // call
+      const out = await evaluateNft(token_uri);
+
+      // verify output - grading
```
Comment (ryanlabouve) [inline src/lib/utils/functions.test.ts:94]: These comments are great. Really help follow what you are testing. 

If you refactored a little moved them to `it` statements statements the test runner would pick them up and give us descriptive failure messages when a tests blew up. But this is also fine.

e.g.

```js
describe('#evaluateNft', () => {
  let examples = [g_lazy_lions_label, g_alpacadabra_label, g_devs_for_revolution_label];

  it('displays the correct grade', () => {
    // loop and test  
  });
  it('displays the correct metadata', () => {
    // loop and test  
  });
});
```

**Comment by ryanlabouve** [comment]
Looking good so far. Let me know when you are ready for review.

File: package-lock.json (hunk 1 above)
Comment (briancmurphy87) [reply package-lock.json]: @ryanlabouve , sorry didn't realize i committed this.  I just checked out the version from `main` and pushed so there should no longer be a file diff for `package-lock.json` in this PR

File: src/lib/utils/functions.test.ts (hunk 2 above)
Comment (briancmurphy87) [reply src/lib/utils/functions.test.ts:87]: noted and like this approach.  js-land is a whole new world to me so keep the pointers coming!

**Comment by briancmurphy87** [comment]
> Looking good so far. Let me know when you are ready for review.

@ryanlabouve , marked as "ready for review".  In the interest of time we can merge this in now and I'll target some of your "refactor" comments with a follow-up PR

File: src/lib/utils/functions.test.ts [hunk 4]
```diff
@@ -1,5 +1,128 @@
...
+      {
+        const expected = g_nft_examples[nft_example]["expected_output"]["metadata"]
+        const realized = out[2];
+
+        // for the keys we have defined check that they align with what is in realized metadat
+        for (const [key, value] of Object.entries(expected)) {
+          expect(value).toEqual(realized[key]);
+        }
+
+      };
```
Comment (johnmosesman) [inline src/lib/utils/functions.test.ts:125]: Is wrapping these statements in { } a stylistic thing for grouping ideas in the tests or do they have a functional use?

**Review by johnmosesman — APPROVED** [review]

**Pull request merged**